account_form holds a bug in interaction tuto

Hi !
In https://github.com/ocsigen/tuto/blob/master/tutos/dev/manual/interaction.wiki (at least from 6.4 version; referenced online there http://ocsigen.org/tuto/6.4/manual/interaction ) :
definition of `account_form` shall take `()` as a parameter (or Form.post_form shall be partially applied, thus we shall remove final parameter () ) to prevent following troubleshouting error building the application:
```
make test.byteeliomdep -server -ppx -package lwt_ppx -package js_of_ocaml-ppx.deriving tuto.ml > _deps/tuto.ml.server
cat _deps/tuto.ml.server > .depend
eliomc -ppx -c -package lwt_ppx -package js_of_ocaml-ppx.deriving  tuto.ml
eliomc -ppx -a -o local/lib/tuto/tuto.cma  \
          _server/tuto.cmo
ocsigenserver  -c local/etc/tuto/tuto-test.conf
ocsigenserver: ocsigen:ocsipersist:dbm: Initializing ...
ocsigenserver: ocsigen:ocsipersist:dbm: Launching a new Ocsidbm process: /home/milan/.opam/doc-genius/lib/ocsigenserver/extensions/ocsidbm on directory local/var/data/tuto/ocsipersist.
ocsigenserver: ocsigen:ocsipersist:dbm: ...Initialization complete
ocsigenserver: ocsigen:config: While parsing config file, tag <host>: No defaulthostname, assuming it is "milan-Aspire-E5-573TG"
ocsigenserver: ocsigen:main: Fatal - Error in configuration file: Error while parsing configuration file: Eliom: while loading local/lib/tuto/tuto.cma: Failure("This function cannot be called here because it needs information about the request or the site.")
Makefile:58: recipe for target 'test.byte' failed
make: *** [test.byte] Error 50
```
And so, account_form shall be called with unit parameter `()` inside registration of service new_user_form_service
Best !

--- a/tutos/6.4/manual/interaction.wiki
+++ b/tutos/6.4/manual/interaction.wiki
@@ -327,7 +327,7 @@ and password to be shown in the URL, we use //hidden parameters//
 these parameters:
 
 <<code language="ocaml"|
-let connection_service = Eliom_service.attach_post
+let connection_service = Eliom_service.create_attached_post
   ~fallback:main_service
   ~post_params:Eliom_parameter.(string "name" ** string "password")
   ()
@@ -337,7 +337,7 @@ let connection_service = Eliom_service.attach_post
 
 Services using the POST HTTP method are created using the function
 <<a_api project="eliom" subproject="server" | val Eliom_service.create >>
-or <<a_api project="eliom" subproject="server" | val Eliom_service.attach_post >>
+or <<a_api project="eliom" subproject="server" | val Eliom_service.create_attached_post >>
 (depending on what type of service you want to create).
 As you can see, there is a major
 difference from the way we created our previous services.
@@ -722,7 +722,7 @@ let redir_service = Eliom_service.create
     ~meth:(Eliom_service.Get Eliom_parameter.unit)
     ()
 
-let service_to_redirect = Eliom_service.attach_post
+let service_to_redirect = Eliom_service.create_attached_post
   ~fallback:redir_service
   ~post_params:(Eliom_parameter.string "something")
   ()
@@ -813,12 +813,12 @@ let new_user_form_service = Eliom_service.create
   ~meth:(Eliom_service.Get Eliom_parameter.unit)
   ()
 
-let create_account_service = Eliom_service.attach_post
+let create_account_service = Eliom_service.create_attached_post
   ~fallback:main_service
   ~post_params:(Eliom_parameter.(string "name" ** string "password"))
   ()
 
-let account_form = Eliom_content.Html.D.(
+let account_form () = Eliom_content.Html.D.(
   Form.post_form ~service:create_account_service
     (fun (name1, name2) ->
       [fieldset
@@ -839,7 +839,7 @@ let _ = Eliom_content.Html.D.(
       Lwt.return
         (html (head (title (txt "")) [])
               (body [h1 [txt "Create an account"];
-                     account_form;
+                     account_form ();
                     ])));
 
   Eliom_registration.Action.register
@@ -910,7 +910,7 @@ new POST attached coservice called {{{account_confirmation_service}}}:
 
 <<code language="ocaml"|
 let account_confirmation_service =
-  Eliom_service.create
+  Eliom_service.create_attached_post
     ~fallback:new_user_form_service
     ~post_params:(Eliom_parameter.(string "name" **  string "password"))
     ()
@@ -928,7 +928,7 @@ creation service:
     ~service:account_confirmation_service
     (fun () (name, pwd) ->
       let create_account_service =
-	Eliom_service.attach_get
+	Eliom_service.create_attached_get
           ~fallback:main_service
           ~get_params:Eliom_parameter.unit
           ~timeout:60.
@@ -1058,8 +1058,8 @@ Modify the connection box this way:
 
 <<code language="ocaml"|
 let connection_box () = Eliom_content.Html.D.(
-  lwt u = Eliom_reference.get username in
-  lwt wp = Eliom_reference.get wrong_pwd in
+  let%lwt u = Eliom_reference.get username in
+  let%lwt wp = Eliom_reference.get wrong_pwd in
   Lwt.return
     (match u with
       | Some s -> div [p [txt "You are connected as "; txt s; ];

--- a/tutos/6.4/manual/tutowidgets.wiki
+++ b/tutos/6.4/manual/tutowidgets.wiki
@@ -70,9 +70,9 @@ let%client mywidget s1 s2 = Eliom_content.Html.D.(
 )
 
 let%client _ =
-  let%lwt _ = Lwt_js_events.onload () in
-  Dom.appendChild
-    (Dom_html.document##.body)
+  let%lwt _ = Js_of_ocaml_lwt.Lwt_js_events.onload () in
+  Js_of_ocaml.Dom.appendChild
+    (Js_of_ocaml.Dom_html.document##.body)
     (Eliom_content.Html.To_dom.of_element (mywidget "Click me" "Hello!"));
   Lwt.return ()
 >>
@@ -117,10 +117,10 @@ or the manual of each of the projects.
   thread to terminate before continuing. {{{e2}}} must ben a Lwt
   thread itself. {{{Lwt.return}}} enables creating an
   already-terminated Lwt thread.
-* {{{Lwt_js_events}}} defines a convenient way to program
+* {{{Js_of_ocaml_lwt.Lwt_js_events}}} defines a convenient way to program
   interface events (mouse, keyboard, ...).
 
-{{{Lwt_js_events.onload}}} is a Lwt thread that waits until the page
+{{{Js_of_ocaml_lwt.Lwt_js_events.onload}}} is a Lwt thread that waits until the page
 is loaded. There are similar functions to wait for other events, e.g.,
 for a click on an element of the page, or for a key press.
 
@@ -132,28 +132,28 @@ Replace function {{{mywidget}}} by the following lines:
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##contains (Js.string "hidden")) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##contains (Js_of_ocaml.Js.string "hidden")) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 let%client mywidget s1 s2 = Eliom_content.Html.D.(
   let button  = div ~a:[a_class ["button"]] [txt s1] in
   let content = div ~a:[a_class ["content"]] [txt s2] in
   Lwt.async (fun () ->
-    Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element button)
+    Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element button)
       (fun _ _ -> switch_visibility content; Lwt.return ()));
   div ~a:[a_class ["mywidget"]] [button; content]
 )
 >>
 
-* Once again, we use {{{Lwt_js_events}}}. Function {{{clicks}}} is
+* Once again, we use {{{Js_of_ocaml_lwt.Lwt_js_events}}}. Function {{{clicks}}} is
   used to bind a handler to clicks on a specific element.
 * Function {{{async}}} runs an {{{Lwt}}} thread asynchronously
   (without waiting for its result).
-* {{{Lwt_js_events.clicks elt f}}} calls function {{{f}}} for each
+* {{{Js_of_ocaml_lwt.Lwt_js_events.clicks elt f}}} calls function {{{f}}} for each
   mouseclick on element {{{elt}}}.
-* {{{Eliom_content.Html.To_dom.of_element}}}, {{{Js.string}}} and {{{Js.to_bool}}} are
+* {{{Eliom_content.Html.To_dom.of_element}}}, {{{Js_of_ocaml.Js.string}}} and {{{Js_of_ocaml.Js.to_bool}}} are
   conversion functions between OCaml values and JS values.
 
 == Third step: Generating the widget on server side ==
@@ -187,10 +187,10 @@ module%server Ex_app =
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="server"|
@@ -199,7 +199,7 @@ let%server mywidget s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button)
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button)
          (fun _ _ -> switch_visibility ~%content; Lwt.return ()))
      : unit)
   ] in
@@ -250,10 +250,10 @@ let%client close_last = ref (fun () -> ())
 
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="shared"|
@@ -262,7 +262,7 @@ let%shared mywidget s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
 	 !close_last ();
 	 close_last := (fun () -> switch_visibility ~%content);
 	 switch_visibility ~%content;
@@ -281,8 +281,8 @@ let%server _ = Eliom_content.Html.D.(
     (fun () () ->
       let _ =
 	[%client
-	    (Dom.appendChild
-	       (Dom_html.document##.body)
+	    (Js_of_ocaml.Dom.appendChild
+	       (Js_of_ocaml.Dom_html.document##.body)
                (Eliom_content.Html.To_dom.of_element (mywidget "Click me" "client side"))
                : unit)
 	] in
@@ -320,10 +320,10 @@ let%server new_set () = [%client ( ref (fun () -> ()) : (unit -> unit) ref)]
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="shared"|
@@ -332,7 +332,7 @@ let%shared mywidget set s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"; "hidden"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
          ! ~%set ();
          ~%set := (fun () -> switch_visibility ~%content);
          switch_visibility ~%content;
@@ -352,8 +352,8 @@ let%server _ = Eliom_content.Html.D.(
       let set1 = new_set () in
       let set2 = new_set () in
       let _ = [%client (
-	Dom.appendChild
-	  (Dom_html.document##.body)
+	Js_of_ocaml.Dom.appendChild
+	  (Js_of_ocaml.Dom_html.document##.body)
 	  (Eliom_content.Html.To_dom.of_element (mywidget ~%set2 "Click me" "client side"))
 	  : unit)
 	      ] in

--- a/tutos/dev/manual/interaction.wiki
+++ b/tutos/dev/manual/interaction.wiki
@@ -327,7 +327,7 @@ and password to be shown in the URL, we use //hidden parameters//
 these parameters:
 
 <<code language="ocaml"|
-let connection_service = Eliom_service.attach_post
+let connection_service = Eliom_service.create_attached_post
   ~fallback:main_service
   ~post_params:Eliom_parameter.(string "name" ** string "password")
   ()
@@ -337,7 +337,7 @@ let connection_service = Eliom_service.attach_post
 
 Services using the POST HTTP method are created using the function
 <<a_api project="eliom" subproject="server" | val Eliom_service.create >>
-or <<a_api project="eliom" subproject="server" | val Eliom_service.attach_post >>
+or <<a_api project="eliom" subproject="server" | val Eliom_service.create_attached_post >>
 (depending on what type of service you want to create).
 As you can see, there is a major
 difference from the way we created our previous services.
@@ -722,7 +722,7 @@ let redir_service = Eliom_service.create
     ~meth:(Eliom_service.Get Eliom_parameter.unit)
     ()
 
-let service_to_redirect = Eliom_service.attach_post
+let service_to_redirect = Eliom_service.create_attached_post
   ~fallback:redir_service
   ~post_params:(Eliom_parameter.string "something")
   ()
@@ -813,12 +813,12 @@ let new_user_form_service = Eliom_service.create
   ~meth:(Eliom_service.Get Eliom_parameter.unit)
   ()
 
-let create_account_service = Eliom_service.attach_post
+let create_account_service = Eliom_service.create_attached_post
   ~fallback:main_service
   ~post_params:(Eliom_parameter.(string "name" ** string "password"))
   ()
 
-let account_form = Eliom_content.Html.D.(
+let account_form () = Eliom_content.Html.D.(
   Form.post_form ~service:create_account_service
     (fun (name1, name2) ->
       [fieldset
@@ -839,7 +839,7 @@ let _ = Eliom_content.Html.D.(
       Lwt.return
         (html (head (title (txt "")) [])
               (body [h1 [txt "Create an account"];
-                     account_form;
+                     account_form ();
                     ])));
 
   Eliom_registration.Action.register
@@ -910,7 +910,7 @@ new POST attached coservice called {{{account_confirmation_service}}}:
 
 <<code language="ocaml"|
 let account_confirmation_service =
-  Eliom_service.create
+  Eliom_service.create_attached_post
     ~fallback:new_user_form_service
     ~post_params:(Eliom_parameter.(string "name" **  string "password"))
     ()
@@ -928,7 +928,7 @@ creation service:
     ~service:account_confirmation_service
     (fun () (name, pwd) ->
       let create_account_service =
-	Eliom_service.attach_get
+	Eliom_service.create_attached_get
           ~fallback:main_service
           ~get_params:Eliom_parameter.unit
           ~timeout:60.
@@ -1058,8 +1058,8 @@ Modify the connection box this way:
 
 <<code language="ocaml"|
 let connection_box () = Eliom_content.Html.D.(
-  lwt u = Eliom_reference.get username in
-  lwt wp = Eliom_reference.get wrong_pwd in
+  let%lwt u = Eliom_reference.get username in
+  let%lwt wp = Eliom_reference.get wrong_pwd in
   Lwt.return
     (match u with
       | Some s -> div [p [txt "You are connected as "; txt s; ];

--- a/tutos/dev/manual/tutowidgets.wiki
+++ b/tutos/dev/manual/tutowidgets.wiki
@@ -70,9 +70,9 @@ let%client mywidget s1 s2 = Eliom_content.Html.D.(
 )
 
 let%client _ =
-  let%lwt _ = Lwt_js_events.onload () in
-  Dom.appendChild
-    (Dom_html.document##.body)
+  let%lwt _ = Js_of_ocaml_lwt.Lwt_js_events.onload () in
+  Js_of_ocaml.Dom.appendChild
+    (Js_of_ocaml.Dom_html.document##.body)
     (Eliom_content.Html.To_dom.of_element (mywidget "Click me" "Hello!"));
   Lwt.return ()
 >>
@@ -117,10 +117,10 @@ or the manual of each of the projects.
   thread to terminate before continuing. {{{e2}}} must ben a Lwt
   thread itself. {{{Lwt.return}}} enables creating an
   already-terminated Lwt thread.
-* {{{Lwt_js_events}}} defines a convenient way to program
+* {{{Js_of_ocaml_lwt.Lwt_js_events}}} defines a convenient way to program
   interface events (mouse, keyboard, ...).
 
-{{{Lwt_js_events.onload}}} is a Lwt thread that waits until the page
+{{{Js_of_ocaml_lwt.Lwt_js_events.onload}}} is a Lwt thread that waits until the page
 is loaded. There are similar functions to wait for other events, e.g.,
 for a click on an element of the page, or for a key press.
 
@@ -132,28 +132,28 @@ Replace function {{{mywidget}}} by the following lines:
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##contains (Js.string "hidden")) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##contains (Js_of_ocaml.Js.string "hidden")) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 let%client mywidget s1 s2 = Eliom_content.Html.D.(
   let button  = div ~a:[a_class ["button"]] [txt s1] in
   let content = div ~a:[a_class ["content"]] [txt s2] in
   Lwt.async (fun () ->
-    Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element button)
+    Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element button)
       (fun _ _ -> switch_visibility content; Lwt.return ()));
   div ~a:[a_class ["mywidget"]] [button; content]
 )
 >>
 
-* Once again, we use {{{Lwt_js_events}}}. Function {{{clicks}}} is
+* Once again, we use {{{Js_of_ocaml_lwt.Lwt_js_events}}}. Function {{{clicks}}} is
   used to bind a handler to clicks on a specific element.
 * Function {{{async}}} runs an {{{Lwt}}} thread asynchronously
   (without waiting for its result).
-* {{{Lwt_js_events.clicks elt f}}} calls function {{{f}}} for each
+* {{{Js_of_ocaml_lwt.Lwt_js_events.clicks elt f}}} calls function {{{f}}} for each
   mouseclick on element {{{elt}}}.
-* {{{Eliom_content.Html.To_dom.of_element}}}, {{{Js.string}}} and {{{Js.to_bool}}} are
+* {{{Eliom_content.Html.To_dom.of_element}}}, {{{Js_of_ocaml.Js.string}}} and {{{Js_of_ocaml.Js.to_bool}}} are
   conversion functions between OCaml values and JS values.
 
 == Third step: Generating the widget on server side ==
@@ -187,10 +187,10 @@ module%server Ex_app =
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="server"|
@@ -199,7 +199,7 @@ let%server mywidget s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button)
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button)
          (fun _ _ -> switch_visibility ~%content; Lwt.return ()))
      : unit)
   ] in
@@ -250,10 +250,10 @@ let%client close_last = ref (fun () -> ())
 
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="shared"|
@@ -262,7 +262,7 @@ let%shared mywidget s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
 	 !close_last ();
 	 close_last := (fun () -> switch_visibility ~%content);
 	 switch_visibility ~%content;
@@ -281,8 +281,8 @@ let%server _ = Eliom_content.Html.D.(
     (fun () () ->
       let _ =
 	[%client
-	    (Dom.appendChild
-	       (Dom_html.document##.body)
+	    (Js_of_ocaml.Dom.appendChild
+	       (Js_of_ocaml.Dom_html.document##.body)
                (Eliom_content.Html.To_dom.of_element (mywidget "Click me" "client side"))
                : unit)
 	] in
@@ -320,10 +320,10 @@ let%server new_set () = [%client ( ref (fun () -> ()) : (unit -> unit) ref)]
 <<code language="ocaml" class="client"|
 let%client switch_visibility elt =
   let elt = Eliom_content.Html.To_dom.of_element elt in
-  if Js.to_bool (elt##.classList##(contains (Js.string "hidden"))) then
-    elt##.classList##remove (Js.string "hidden")
+  if Js_of_ocaml.Js.to_bool (elt##.classList##(contains (Js_of_ocaml.Js.string "hidden"))) then
+    elt##.classList##remove (Js_of_ocaml.Js.string "hidden")
   else
-    elt##.classList##add (Js.string "hidden")
+    elt##.classList##add (Js_of_ocaml.Js.string "hidden")
 
 >>
 <<code language="ocaml" class="shared"|
@@ -332,7 +332,7 @@ let%shared mywidget set s1 s2 = Eliom_content.Html.D.(
   let content = div ~a:[a_class ["content"; "hidden"]] [txt s2] in
   let _ = [%client
     (Lwt.async (fun () ->
-       Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
+       Js_of_ocaml_lwt.Lwt_js_events.clicks (Eliom_content.Html.To_dom.of_element ~%button) (fun _ _ ->
          ! ~%set ();
          ~%set := (fun () -> switch_visibility ~%content);
          switch_visibility ~%content;
@@ -352,8 +352,8 @@ let%server _ = Eliom_content.Html.D.(
       let set1 = new_set () in
       let set2 = new_set () in
       let _ = [%client (
-	Dom.appendChild
-	  (Dom_html.document##.body)
+	Js_of_ocaml.Dom.appendChild
+	  (Js_of_ocaml.Dom_html.document##.body)
 	  (Eliom_content.Html.To_dom.of_element (mywidget ~%set2 "Click me" "client side"))
 	  : unit)
 	      ] in

